app/main/views/auth.py

```diff
@@ -87,6 +88,13 @@ def match_idp_email(idp, email_address):
     return re.match(idp['email_pattern'], email_address)
 
 
+def dept_from_idp_name(idp_name):
+    idp = [idp['name'] for idp in idp_profiles if idp_name == idp['idp_name']]
+    if idp:
+        return ' or '.join(idp)
+    return None
```

Comment:
You could just `return ' or '.join(dept_names)` - if there are no matching `idp_profiles` then this will be an empty string. This equals `False`, so instead of testing for `None` you can do:
```python
assert not dept_from_idp_name('unknown')
```